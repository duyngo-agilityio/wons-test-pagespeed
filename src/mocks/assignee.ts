import { TRole } from '@/types';

export const MOCK_ASSIGNEES = {
  data: [
    {
      id: 1,
      attributes: {
        name: 'John Doe',
        avatar: '/avatar1.jpg',
        email: 'john@example.com',
        username: 'john_doe',
        fullName: 'John Doe',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
    {
      id: 2,
      attributes: {
        name: 'Jane Smith',
        avatar: '/avatar2.jpg',
        email: 'jane@example.com',
        username: 'jane_smith',
        fullName: 'Jane Smith',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
  ],
};
