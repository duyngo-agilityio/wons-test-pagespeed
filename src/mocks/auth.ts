// Types
import { ISignUpFormData } from '@/types';

export const SIGN_UP_FORM_DATA_MOCK: ISignUpFormData = {
  fullName: 'John Doe',
  username: 'johndoe',
  email: 'john.doe@example.com',
  password: 'Abcd@0987',
};

export const MOCK_SUCCESS_SIGN_UP_RES = {
  data: { user: { ...SIGN_UP_FORM_DATA_MOCK, id: '1' } },
  totalCount: 0,
};
