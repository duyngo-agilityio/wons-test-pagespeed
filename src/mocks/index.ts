export * from './customer';
export * from './recentServices';
export * from './tabs';
