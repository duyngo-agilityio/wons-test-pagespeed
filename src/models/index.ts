export * from './customer';
export * from './invoice';
export * from './product';
export * from './customer';
export * from './invoice';
export * from './statistic';
export * from './user';
export * from './calendar';
