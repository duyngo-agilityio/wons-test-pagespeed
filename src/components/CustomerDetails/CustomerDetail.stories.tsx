import type { Meta, StoryObj } from '@storybook/react';

// Mocks
import { CUSTOMER_MOCK } from '@/mocks';

// Components
import CustomerDetails from '.';

const meta: Meta<typeof CustomerDetails> = {
  title: 'Components/CustomerDetails',
  component: CustomerDetails,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
};

export default meta;
type Story = StoryObj<typeof CustomerDetails>;

export const Default: Story = {
  args: {
    customer: CUSTOMER_MOCK[0],
    isLoading: false,
  },
};
