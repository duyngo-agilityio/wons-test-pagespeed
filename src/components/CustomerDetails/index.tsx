import { memo } from 'react';
import isEqual from 'react-fast-compare';

// Models
import { ICustomer } from '@/models';

// Components
import CustomerDetailsHeader from './header';
import CustomerDetailsBody from './body';

interface ICustomerDetails {
  customer: ICustomer;
}

const CustomerDetails = ({ customer }: ICustomerDetails) => {
  const {
    avatar = '',
    firstName = '',
    lastName = '',
    fullName = '',
    job = '',
    email = '',
    phone = '',
    address = '',
  } = customer ?? {};

  return (
    <div className="min-h-full max-w-[302px] bg-white dark:bg-gray-400 py-[62px] px-6">
      <CustomerDetailsHeader
        avatar={avatar}
        firstName={firstName}
        lastName={lastName}
        fullName={fullName}
        job={job}
      />
      <CustomerDetailsBody email={email} phone={phone} address={address} />
    </div>
  );
};

export default memo(CustomerDetails, isEqual);
