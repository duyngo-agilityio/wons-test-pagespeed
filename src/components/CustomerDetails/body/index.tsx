import { memo } from 'react';

// Components
import { EmailIcon, FaLocationDot, ImPhone, Text } from '@/components/common';

// Utils
import { formatPhoneNumberTyping } from '@/utils';

interface ICustomerDetailsBody {
  email: string;
  phone: string;
  address: string;
}

const CustomerDetailsBody = ({
  email,
  phone,
  address,
}: ICustomerDetailsBody) => (
  <div className="flex flex-col gap-2.5 mt-10">
    <Text text="Contact info" className="capitalize font-medium" size="xxl" />
    <div>
      <div className="flex items-center gap-2 border-b-1 border-blue-800/10 dark:border-white/10 py-4">
        <EmailIcon
          className="text-blue-800/30 dark:text-white/30"
          fontSize={16}
        />
        <Text text={email} size="xl" className="opacity-70 font-medium" />
      </div>
      <div className="flex items-center gap-2 border-b-1 border-blue-800/10 dark:border-white/10 py-4">
        <ImPhone
          className="text-blue-800/30 dark:text-white/30"
          fontSize={16}
        />
        <Text
          text={formatPhoneNumberTyping(phone)}
          size="xl"
          className="opacity-70 font-medium"
        />
      </div>
      <div className="flex items-center gap-2 py-4">
        <FaLocationDot
          className="text-blue-800/30 dark:text-white/30"
          fontSize={16}
        />
        <Text text={address} size="xl" className="opacity-70 font-medium" />
      </div>
    </div>
  </div>
);

export default memo(CustomerDetailsBody);
