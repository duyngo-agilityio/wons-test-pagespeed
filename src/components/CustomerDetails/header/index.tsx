import { memo } from 'react';

// Components
import { ImageFallback, Text } from '@/components/common';

interface ICustomerDetailsHeader {
  avatar: string;
  firstName: string;
  lastName: string;
  fullName: string;
  job: string;
}

const CustomerDetailsHeader = ({
  avatar,
  firstName,
  fullName,
  job,
  lastName,
}: ICustomerDetailsHeader) => (
  <div className="flex flex-col items-center gap-4 border-b-1 border-blue-800/10 dark:border-white/10 pb-7.5">
    <div className="relative w-[70px] h-[70px]">
      <ImageFallback
        fill
        src={avatar}
        alt={`${firstName}-${lastName}`}
        className="object-cover rounded-full"
      />
    </div>
    <div className="flex flex-col items-center">
      <Text text={fullName} size="3xl" className="font-bold" />
      {job && <Text text={job} size="md" />}
    </div>
  </div>
);

export default memo(CustomerDetailsHeader);
