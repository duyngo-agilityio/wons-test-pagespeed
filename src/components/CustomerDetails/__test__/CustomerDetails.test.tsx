import { CUSTOMER_MOCK } from '@/mocks';
import CustomerDetails from '../index';
import { ICustomer } from '@/models';

describe('CustomerDetails', () => {
  const renderComponent = (
    props?: Record<string, boolean | ICustomer | undefined>,
  ) =>
    testLibJestUtils.render(
      <CustomerDetails customer={CUSTOMER_MOCK[0]} {...props} />,
    );
  it('should match snapshot', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with loading', () => {
    const { container } = renderComponent({
      isLoading: true,
    });

    expect(container).toMatchSnapshot();
  });
});
