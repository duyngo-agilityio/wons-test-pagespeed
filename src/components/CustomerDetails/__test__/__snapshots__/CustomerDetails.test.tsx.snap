// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDetails should match snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="min-h-full max-w-[302px] bg-white dark:bg-gray-400 py-[62px] px-6"
    >
      <div
        class="flex flex-col items-center gap-4 border-b-1 border-blue-800/10 dark:border-white/10 pb-7.5"
      >
        <div
          class="relative w-[70px] h-[70px]"
        >
          <img
            alt="John-Doe"
            class="object-cover rounded-full h-full"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="70px"
            src="http://localhost/_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fdev.perlinks.com%2Fapi%2Fget-image%2FWGpHZ&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          />
        </div>
        <div
          class="flex flex-col items-center"
        >
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-3xl font-bold"
          >
            John Doe
          </p>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-md "
          >
            Software Engineer
          </p>
        </div>
      </div>
      <div
        class="flex flex-col mt-10"
      >
        <p
          class="font-dm-sans text-blue-800 dark:text-white text-xxl capitalize font-medium"
        >
          Contact info
        </p>
        <div
          class="flex items-center gap-2 mt-[10px] border-b-1 border-blue-800/10 dark:border-white/10 py-4"
        >
          <svg
            class="text-blue-800/30 dark:text-white/30"
            color="#3A36DB"
            font-size="16"
            height="14"
            viewBox="0 0 16 14"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.6192 0C12.6622 0 13.6656 0.412222 14.4037 1.15189C15.1426 1.89 15.5556 2.88556 15.5556 3.92778V10.0722C15.5556 12.2422 13.79 14 11.6192 14H3.93556C1.76478 14 0 12.2422 0 10.0722V3.92778C0 1.75778 1.757 0 3.93556 0H11.6192ZM12.8567 5.08673L12.9189 5.02451C13.1048 4.79895 13.1048 4.47229 12.9103 4.24673C12.8022 4.13084 12.6537 4.06007 12.4989 4.04451C12.3356 4.03595 12.18 4.09118 12.0626 4.20007L8.55556 7.00007C8.10444 7.37418 7.45811 7.37418 7 7.00007L3.5 4.20007C3.25811 4.02118 2.92367 4.04451 2.72222 4.25451C2.51222 4.46451 2.48889 4.79895 2.667 5.03229L2.76889 5.1334L6.30778 7.89451C6.74333 8.23673 7.27144 8.4234 7.82444 8.4234C8.37589 8.4234 8.91333 8.23673 9.34811 7.89451L12.8567 5.08673Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
          >
            john.doe@example.com
          </p>
        </div>
        <div
          class="flex items-center gap-2 border-b-1 border-blue-800/10 dark:border-white/10 py-4"
        >
          <svg
            class="text-blue-800/30 dark:text-white/30"
            fill="currentColor"
            font-size="16"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 10c-1 1-1 2-2 2s-2-1-3-2-2-2-2-3 1-1 2-2-2-4-3-4-3 3-3 3c0 2 2.055 6.055 4 8s6 4 8 4c0 0 3-2 3-3s-3-4-4-3z"
            />
          </svg>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
          >
            (123) 456-7890
          </p>
        </div>
        <div
          class="flex items-center gap-2 py-4"
        >
          <svg
            class="text-blue-800/30 dark:text-white/30"
            fill="currentColor"
            font-size="16"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
            />
          </svg>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
          >
            123 Main St, Anytown, USA
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CustomerDetails should match snapshot with loading 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="fixed inset-0 flex items-center justify-center bg-black bg-opacity-70 z-50"
    >
      <div
        aria-label="Loading"
        class="relative inline-flex flex-col gap-2 items-center justify-center"
      >
        <div
          class="relative flex w-10 h-10"
        >
          <i
            class="absolute w-full h-full rounded-full animate-spinner-ease-spin border-solid border-t-transparent border-l-transparent border-r-transparent border-3 border-b-blue-500"
          />
          <i
            class="absolute w-full h-full rounded-full opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-l-transparent border-r-transparent border-3 border-b-blue-500"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
